docs/development/versioning.md

```diff
@@ -0,0 +1,107 @@
+
+## Definition of "Public API"
+
+The [**files**](../../schema/files) and [**objects**](../../schema/objects/) portions of the OCF
+Schema are considered the "Public API". When using the schema to generate code, validate JSON, etc.,
+users of the schema should use these schema definitions.
+
+All other portions of the schema ([**enums**](../../schema/enums/),
+[**primitives**](../../schema/primitives/), etc.) are considered implementation details, and should
+not be referenced directly, except from within the schema itself.
+
+### `$id` Referential Stability
+
+When using _released versions_ of the schema, `$id` attributes for **files** and **objects** are
+guaranteed to be stable within a MAJOR version. For example, if you create tooling referencing the
+schema URIs such as:
+
+    https://schema.opencaptablecoalition.com/v/1.1.0/files/OCFManifestFile.schema.json
+    https://schema.opencaptablecoalition.com/v/1.1.0/objects/Valuation.schema.json
+
+you are guaranteed that all MINOR and PATCH versions in MAJOR version 1 will have
+_backwards-compatible_ schemas at:
+
+    https://schema.opencaptablecoalition.com/v/1.Y.Z/files/OCFManifestFile.schema.json
+    https://schema.opencaptablecoalition.com/v/1.Y.Z/objects/Valuation.schema.json
+
+On the other hand, all other files within the schema are free to be re-arranged, renamed, deleted,
+etc. as desired by the maintainers to best support the schema at large. In other words, if you
+choose to directly reference the schema URI:
+
+    https://schema.opencaptablecoalition.com/v/1.1.0/enums/InterestPayoutType.schema.json
+
+you **are NOT** guaranteed that updates within MAJOR version 1 will have a backwards-compatible
+schema, or any schema at all, at:
+
+    https://schema.opencaptablecoalition.com/v/1.Y.Z/enums/InterestPayoutType.schema.json
+
+## Methods to Avoid Breaking Changes
+
+When making changes to the schema files, contributors should generally avoid breaking changes. If
+one believes a breaking change is necessary to implement a feature or fix a bug, consult the
+Technical Working Group before submitting the pull request.
+
+_Any change that would cause a previously valid JSON representation of a **file** or **object** to
+fail schema validation is a breaking change._
+
+> **Exception:** A pull request submitted to _fix_ a breaking change that was mistakenly introduced
+> is allowed to be a "breaking" change when compared to the flawed releases, without incrementing
+> the MAJOR version number, as long as it maintains compatibility with older releases. The faulty
+> releases will be deprecated and/or unpublished.
+
+### Renaming a Required Property
+
+If we determine a required property needs to be renamed, we still have to support the old property
+name until a new MAJOR version of the schema is released. One specific reason we have renamed a
+property was to support multiple values where only one value was previously supported.
```

Comment:
@pjohnmeyer Any thoughts on how a replaced field should be tagged as "old"/"deprecated" so that it's clear what needs to be cleaned up when a new MAJOR version of the schema is released? I've come across a few instances where deprecated values are called out in schema descriptions. If that works, it could be helpful to formalize that process here as well.
eg `"description": "This is done to avoid a breaking change as we work towards a bigger restructure of the equity types in v2.0.0. `TX_PLAN_SECURITY_ACCEPTANCE` will be deprecated in v2.0.0"`

Reply:
In the Wiki tab in GitHub, John had started a document called "Planned 2.0 Rework" with a running list of breaking changes. There's also a note about how we started using `$comment` to denote some v2 changes

Reply:
`$comment` works for some things but not for others -- but it could definitely work for a replaced field, as seen in:

https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/blob/24199843757bb78db87ad90e50d7a7f98d4ae37f/schema/objects/transactions/issuance/EquityCompensationIssuance.schema.json#L38-L42

I wonder, can we use the `deprecated` keyword as [defined in JSON Schema 2019-09](https://json-schema.org/draft/2019-09/json-schema-validation#rfc.section.9.3) even though we're currently on draft-7? Will that break existing tooling, or will it just sit passively until we move to a newer version of JSON Schema? If it can sit passively that would actually be my preference.